modules/webbrowser/include/modules/webbrowser/io/json/minmaxpropertyjsonconverter.h

```diff
@@ -0,0 +1,112 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/webbrowser/webbrowsermoduledefine.h>
+#include <inviwo/core/properties/minmaxproperty.h>
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace glm {
+template <typename T, precision P, template <typename, precision> class VecType>
+void from_json(const nlohmann::json& j, VecType<T, P>& v) {
+    for (int i = 0; i < v.length(); i++) {
+        v[i] = j[i];
+    }
+};
+
+template <typename T, precision P, template <typename, precision> class VecType>
+void to_json(nlohmann::json& j, const VecType<T, P>& v) {
+    for (int i = 0; i < v.length(); i++) {
+        j[i] = v[i];
+    }
+};
+}  // namespace glm
+
+namespace inviwo {
+
+/**
+ * Converts an MinMaxProperty to a JSON object.
+ * Produces layout according to the members of MinMaxProperty:
+ * { {"start": val}, {"end": val},
+ *   {"minValue": minVal}, {"maxValue": maxVal},
+ *   {"increment": increment}, {"minSeparation": minSep}
+ * }
+ * @see MinMaxProperty
+ *
+ * Usage example:
+ * \code{.cpp}
+ * MinMaxProperty<double> p;
+ * json j = p;
+ * \endcode
+ */
+template <typename T>
+void to_json(json& j, const MinMaxProperty<T>& p) {
```

Comment:
Still need a factory here for this to work with third party proprties...

Reply:
on the todo :)
Probably need a widget for each property anyway so there is one type of factory already. But sure, I’m working on it

Reply:
Check out latest commit. Comments on interface?

Reply:
Looks reasonable I think

Reply:
Done